File: pkg/src/contract/Ast/AstFileSyntaxErrorEvent/AstFileSyntaxErrorEvent.php
<?php

declare (strict_types=1);
namespace Qossmic\Deptrac\Contract\Ast;

use DEPTRAC_INTERNAL\Symfony\Contracts\EventDispatcher\Event;
/**
 *
 */
final class AstFileSyntaxErrorEvent extends Event
{
    public function __construct(public readonly string $file, public readonly string $syntaxError)
    {
    }
}

File: pkg/src/contract/ast/ast_file_syntax_error_event.go
package ast

// AstFileSyntaxErrorEvent - Event triggered when parsing the AST failed on syntax error in the PHP file.
type AstFileSyntaxErrorEvent struct {
	File        string
	SyntaxError string
}

func NewAstFileSyntaxErrorEvent(file string, syntaxError string) *AstFileSyntaxErrorEvent {
	return &AstFileSyntaxErrorEvent{File: file, SyntaxError: syntaxError}
}
